libflux/src/flux/lib.rs

```diff
@@ -223,3 +223,109 @@ pub unsafe extern "C" fn flux_error_str(err: *mut flux_error_t) -> *mut c_char {
 pub unsafe extern "C" fn flux_free(err: *mut c_void) {
     Box::from_raw(err);
 }
+
+/// flux_merge_ast_pkg_files merges the files of a given in_pkg into the file
+/// vector of a given out_pkg. If there are problems with the package identifiers,
+/// then a panic may occur.
+#[no_mangle]
+pub unsafe extern "C" fn flux_merge_ast_pkg_files(
```

Comment:
Ah, I think I understand my confusion. A panic from Rust will not result in a panic in Go. A panic in Rust will just corrupt the memory of the Go process(which may or may not panic the Go side of things). I am not sure what will happen then. We cannot panic from this Rust function. We must ensure that all errors are returned explicitly. Perhaps this is related to the bug you are looking for.

Reply:
Yeah, there aren't any panics anymore. I'm just propagating errors now. I was panicking originally until I started trying to glue the Go and Rust sides together.